extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -158,7 +157,14 @@ public void startRecoveryService(final TransactionManagerConfiguration transacti
     public void handleShutdown(ShutdownContext context, TransactionManagerConfiguration transactions) {
         context.addLastShutdownTask(() -> {
             if (transactions.enableRecovery) {
-                RecoveryManager.manager().terminate(true);
+                try {
+                    QuarkusRecoveryService.getInstance().stop();
+                } catch (Exception e) {
+                    // the recovery manager throws IllegalStateException if it has already been shutdown
+                    log.warn("The recovery manager has already been shutdown", e);
```

Comment:
@yrodiere The signature of the method that throws the IllegalStateException is `public void stop() throws Exception` [1] so that's what I need to catch .

Note that `QuarkusRecoveryService extends RecoveryManagerService` and RecoveryManagerService is a narayana integration class. In my response to Guillaume I was just pointing out what the actual exception thrown is, I gave links to the code in my response to him.

[1] https://github.com/jbosstm/narayana/blob/6.0.1.Final/ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/jta/RecoveryManagerService.java#L69

Reply:
I think I get what you mean.

But you only know that `IllegalStateException` is about the recovery manager already being shut down.

Any other exception would be unexpected, so even if you have to catch them, you probably shouldn't log a warning saying the recovery manager has already been shut down... ?